libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -288,15 +293,50 @@ MQTTStatus_t MQTT_SerializeUnsubscribe( const MQTTSubscribeInfo_t * const pSubsc
                                         size_t remainingLength,
                                         const MQTTFixedBuffer_t * const pBuffer );
 
+/**
+ * @brief Get the size and Remaining Length of an MQTT PUBLISH packet.
+ *
+ * @param[in] pPublishInfo MQTT PUBLISH packet parameters.
+ * @param[out] pRemainingLength The Remaining Length of the MQTT PUBLISH packet.
+ * @param[out] pPacketSize The total size of the MQTT PUBLISH packet.
+ *
+ * @return #MQTTBadParameter if the packet would exceed the size allowed by the
+ * MQTT spec or if invalid parameters are passed; #MQTTSuccess otherwise.
+ */
 MQTTStatus_t MQTT_GetPublishPacketSize( const MQTTPublishInfo_t * const pPublishInfo,
                                         size_t * const pRemainingLength,
                                         size_t * const pPacketSize );
 
+/**
+ * @brief Serialize an MQTT PUBLISH packet in the given buffer.
+ *
+ * @param[in] pPublishInfo MQTT PUBLISH packet parameters.
+ * @param[in] packetId packet ID generated by #MQTT_GetPacketId.
+ * @param[in] remainingLength Remaining Length provided by #MQTT_GetConnectPacketSize.
+ * @param[out] pBuffer Buffer for packet serialization.
+ *
+ * @return #MQTTNoMemory if pBuffer is too small to hold the MQTT packet;
+ * #MQTTBadParameter if invalid parameters are passed;
+ * #MQTTSuccess otherwise.
+ */
 MQTTStatus_t MQTT_SerializePublish( const MQTTPublishInfo_t * const pPublishInfo,
                                     uint16_t packetId,
                                     size_t remainingLength,
                                     const MQTTFixedBuffer_t * const pBuffer );
 
+/**
+ * @brief Serialize an MQTT PUBLISH packet header in the given buffer.
```

Comment:
Can we provide some rationale why we have two different functions - `MQTT_SerializePublish` and `MQTT_SerializePublishHeader`. It seems that it is to avoid copying of MQTT payload into the network buffer. This is a very good optimization especially for use cases where people are trying to send relatively larger payloads (like images) and therefore, we should document which one should be used when.